lib/link_monitor/upsert_resource_monitor.rb

```diff
@@ -0,0 +1,42 @@
+module LinkMonitor
+  class UpsertResourceMonitor
+    def initialize(links:, service:, resource_type:, resource_id:)
+      @links = links
+      @service = service
+      @resource_type = resource_type
+      @resource_id = resource_id
+    end
+
+    def call
+      monitor = ResourceMonitor.find_or_create_by(service: service, resource_type: resource_type, resource_id: resource_id)
+
+      manage_links(monitor) if monitor.valid?
+
+      monitor
+    end
+
+  private
+
+    attr_reader :links, :service, :resource_type, :resource_id
+
+    def create_links(monitor)
+      links.each do |link|
+        monitored_link = Link.find_or_create_by(uri: link)
+        monitor.monitor_links.find_or_create_by(link: monitored_link)
```

Comment:
Is there any eventuality that would require a link to be updated during this call? We currently handle inserting and deleting, but not updating.